File: backend/src/main/java/kr/co/programmers/collabond/api/user/domain/RefreshToken.java
package kr.co.programmers.collabond.api.user.domain;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import kr.co.programmers.collabond.shared.domain.OnlyUpdatedEntity;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class RefreshToken extends OnlyUpdatedEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @NotNull
    @Size(max = 500)
    @Column(unique = true)
    private String token;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private TokenStatus status = TokenStatus.VALID;

    @Builder
    private RefreshToken(User user, String token) {
        this.user = user;
        this.token = token;
    }

    public void updateStatus(TokenStatus status) {
        this.status = status;
    }
}

File: backend/src/main/java/kr/co/programmers/collabond/api/user/domain/RefreshToken.kt
package kr.co.programmers.collabond.api.user.domain

import jakarta.persistence.*
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size
import kr.co.programmers.collabond.shared.domain.OnlyUpdatedEntity

@Entity
class RefreshToken(
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    var user: User,

    @Column(unique = true)
    @field:Size(max = 500)
    @field:NotNull
    var token: String
) : OnlyUpdatedEntity() {
    @field:NotNull
    @Enumerated(EnumType.STRING)
    var status: TokenStatus = TokenStatus.VALID

    fun updateStatus(newStatus: TokenStatus) {
        status = newStatus
    }
}
